cvmfs/network/sink_mem.h

```diff
@@ -0,0 +1,96 @@
+/**
+ * This file is part of the CernVM File System.
+ */
+
+#ifndef CVMFS_NETWORK_SINK_MEM_H_
+#define CVMFS_NETWORK_SINK_MEM_H_
+
+#include <cstring>
+
+#include "sink.h"
+#include "util/posix.h"
+#include "util/smalloc.h"
+
+namespace cvmfs {
+
+class MemSink : public Sink {
+ public:
+  MemSink() : size_(0), pos_(0), data_(NULL), is_owner_(true) {}
+  explicit MemSink(size_t size) : size_(size), pos_(0) {
+    data_ = static_cast<char *>(smalloc(size));
+    is_owner_ = true;
+  }
+
+  virtual ~MemSink() { if (is_owner_ && data_) { free(data_); } }
+
+  virtual int64_t Write(const void *buf, uint64_t sz) {
+    if (pos_ + sz > size_) {
+      return static_cast<int64_t>(pos_ + sz - size_);
```

Comment:
Let's return a `write` errno here, I'd suggest
```suggestion
      return -ENOSPC;
```